pupynere.py

```diff
@@ -376,16 +376,16 @@ def flush(self):
         sync : Identical function
 
         """
-        if hasattr(self, 'mode') and self.mode is 'w':
+        if hasattr(self, 'mode') and self.mode == 'w':
             self._write()
     sync = flush
 
     def recvars(self):
-        return OrderedDict( filter(lambda (k, v): v.isrec, self.variables.items()) )
+        return OrderedDict( filter(lambda kv: kv[1].isrec, self.variables.items()) )
```

Comment:
I can't remember the exact API for `self.variables`, but does it have an `values()` attribute? This would read slightly cleaner as:
`filter(lambda v: v.isrec, self.variables.values())`